cli/wallet.go

```diff
@@ -634,14 +636,21 @@ var walletMarketWithdraw = &cli.Command{
 			return err
 		}
 
-		var withdrawn abi.TokenAmount
-		if err := withdrawn.UnmarshalCBOR(bytes.NewReader(wait.Receipt.Return)); err != nil {
+		nv, err := api.StateNetworkVersion(ctx, wait.TipSet)
+		if err != nil {
 			return err
 		}
 
-		fmt.Printf("Successfully withdrew %s FIL\n", withdrawn)
-		if withdrawn != amt {
-			fmt.Printf("Note that this is less than the requested amount of %s FIL\n", amt)
+		if nv >= network.Version14 {
+			var withdrawn abi.TokenAmount
+			if err := withdrawn.UnmarshalCBOR(bytes.NewReader(wait.Receipt.Return)); err != nil {
+				return err
+			}
+
+			fmt.Printf("Successfully withdrew %s FIL\n", withdrawn)
+			if withdrawn != amt {
+				fmt.Printf("Note that this is less than the requested amount of %s FIL\n", amt)
```

Comment:
same, we are checking if the balance is < requested.  if so, we will return an error directly and wont get here

https://github.com/filecoin-project/lotus/blob/1d2f84322d4fc1c0d5357476aa693c309097a8c8/cli/wallet.go#L612-L614